main.js

```diff
@@ -1,3 +1,53 @@
+// Selected currency's USD value
+// Defaults to USD if not specified
+let currency = {
+  exchangeRateToUSD: 1,
+  code: "USD",
+};
+
+/**
+ *
+ * @param {string} currencyCode
+ * @returns currency exchange rate from USD to currency.
+ */
+async function getExchangeRateFromUSDToCurrency(currencyCode) {
+  const response = await fetch(
+    `https://cdn.jsdelivr.net/gh/fawazahmed0/currency-api@1/latest/currencies/usd/${currencyCode}.json`
+  );
+  const data = await response.json();
+  return data[currencyCode];
+}
+
+/**
+ * Fetches and updates the currency object with latest values.
+ * @param {string} currencyCode
+ */
+async function updateCurrency(currencyCode) {
+  const conversionRateUSD = await getExchangeRateFromUSDToCurrency(
+    currencyCode
+  );
+  currency.code = currencyCode;
+  currency.exchangeRateToUSD = conversionRateUSD;
+}
+
+chrome.storage.local.get(["currencyCode"]).then(async (result) => {
+  await updateCurrency(result.currencyCode);
```

Comment:
not a good strategy to fetch exchange rates on every page reload. Suggestion: cache it for a day.